app/controllers/update_game.rb

```diff
@@ -1,23 +1,49 @@
+class UpdateGame
+  def update(current_game, position)
+    board = current_game.board
+    player = current_game.player
+    opponent = current_game.opponent
+    @current_player = current_game.current_player
 
-  class UpdateGame
-    def update(current_game,position)
-      board = current_game.board
-      player = current_game.player
-      opponent = current_game.opponent
-      @current_player = current_game.current_player
+    toggle = TicTacToeGame::Toggle.new(player)
+    @game = TicTacToeGame::Game.new(board, player, toggle)
+
+    return false unless board[position] == "-"
+    game_move = GameMove.new
+    game_move.all_moves(@game, opponent, @current_player, position, current_game)
+    current_game.player = toggle.current_turn
+    current_game.board = @game.board
 
-      toggle = TicTacToeGame::Toggle.new(player)
-      @game = TicTacToeGame::Game.new(board, player, toggle)
-  
-      return false unless board[position] == "-" 
-      GameMove.new.all_moves(@game, opponent, @current_player, position, current_game)
+    if !@game.end? && current_game.current_player == "computer"
+      @current_player = current_game.current_player
+      game_move.all_moves(@game, opponent, @current_player, position, current_game)
       current_game.player = toggle.current_turn
       current_game.board = @game.board
```

Comment:
Line 13-15 are duplicated with line 6 + line 13-15. How can we remove the duplication here?